Draft/draft-ietf-rtcweb-stun-consent-freshness.xml

```diff
@@ -378,16 +391,25 @@
       an API hook that generates an event when consent has expired for a given
       5-tuple, meaning that transmission of data has ceased. This could
       indicate what application data is affected, such as media or data
-      channels.</t>
+      channels. The circuit breaker algorithm discussed in section 7.1 of
+      <xref target="I-D.ietf-rtcweb-rtp-usage"></xref> could also be one of
+      the reasons for ceasing transmission of media and to notify the
+      application about the loss of ability to transmit data.</t>
```

Comment:
It was included to address the following comment from David:


> > >> [A.1.5 - network impact]

> > >>

> > >> There is an obvious fault condition, namely that consent is lost

> > >> or revoked causing immediate cessation of traffic.  While the

> > >> details depend on the environment in which this mechanism is

> > >> used, it'd be helpful to add a sentence or two on reporting of

> > >> the state of STUN consent-based connectivity and how that

> > >> reporting should or may relate to reporting of the state of other

> > >> forms of connectivity (e.g., TCP,

> SRTP/SRTCP) that are mentioned in this draft.

> >

> > We specifically discussed in WG about how applications should handle

> > changes in Consent and it was decided to keep it outside the scope

> > of this draft. Para 3 of introduction already has a text that says the same.

> > There can be many possibilities if consent is lost or failed

> > depending on what the application wants.

>

> Ok, but .. this is not about "how applications should handle" loss of

> consent; it's about "how implementations should report" loss of

> consent.  Part of this is already in Section 7, which indicates how

> the application learns that consent has expired.  It would suffice to

> add a sentence added to that section to indicate that there are other

> reasons for loss of ability to transmit data, and include an example

> of how at least one other is reported to applications.

-Tiru

From: Martin Thomson [mailto:notifications@github.com]
Sent: Thursday, May 28, 2015 9:48 PM
To: tireddy/STUNConsent
Cc: Tirumaleswar Reddy (tireddy)
Subject: Re: [STUNConsent] Opsdir and GenArt review (#1)


In Draft/draft-ietf-rtcweb-stun-consent-freshness.xml<https://github.com/tireddy/STUNConsent/pull/1#discussion_r31250824>:

> @@ -378,16 +391,25 @@

>        an API hook that generates an event when consent has expired for a given

>        5-tuple, meaning that transmission of data has ceased. This could

>        indicate what application data is affected, such as media or data

> -      channels.</t>

> +      channels. The circuit breaker algorithm discussed in section 7.1 of

> +      <xref target="I-D.ietf-rtcweb-rtp-usage"></xref> could also be one of

> +      the reasons for ceasing transmission of media and to notify the

> +      application about the loss of ability to transmit data.</t>

Why did you agree to including mention of circuit breakers? They are irrelevant here. And potentially confusing.

—
Reply to this email directly or view it on GitHub<https://github.com/tireddy/STUNConsent/pull/1/files#r31250824>.